### PR TITLE
Cluster with two or more replicas never starts after Spec.Container properties changes

### PR DESCRIPTION
**Pull request by dmvolod**
This is base changes for switching form static source to function provided and for testing with our CI

### CODE REVIEW

File: build/run-tests.sh [hunk 1]
```diff
@@ -5,4 +5,4 @@ KUBECONFIG=${1-openshift.local.clusterup/openshift-apiserver/admin.kubeconfig}
 echo "Using KUBECONFIG '$KUBECONFIG'"
 
 go clean -testcache ./test/e2e
-go test -v ./test/e2e -timeout 45m
+go test -v ./test/e2e -run TestContainerCPUUpdateWithTwoReplicas -timeout 45m
```
Comment (Crumby) [inline build/run-tests.sh]: Isn't this used to run all the tests via Travis CI? This change would drop all but one this one change. (Also the file is named `run-tests`)

Comment (dmvolod) [reply build/run-tests.sh]: @Crumby , yes, you are right, but it just for one issue reproducer. I will drop -run when this issue will be fixed, but I'm thinking about it nature.

**Comment by dmvolod** [comment]
@rigazilla , it's a very strange, but reproducer looks like fails after fix

**Comment by rigazilla** [comment]
@dmvolod the test seems to fail during the first cluster startup:

https://github.com/dmvolod/infinispan-operator/blob/ac727d307c1c73e08820d4a989253853d035c523/test/e2e/main_test.go#L199

I've added a commit, but I'm not sure it's the fix... let's see CI

**Comment by dmvolod** [comment]
Thanks, @rigazilla , I'm testing now on my local env.

**Comment by dmvolod** [comment]
Oh, @rigazilla excellent. Now all are working fine!!!
What do you think about to keep this test in e2e set?

**Comment by rigazilla** [comment]
@dmvolod looks good to me!
Can you squash the commits? I can merge then

**Comment by dmvolod** [comment]
Done, @rigazilla 
Please merge, thanks

**Comment by rigazilla** [comment]
I've squashed my commit also

**Comment by rigazilla** [comment]
merged. Thanks @dmvolod , great job!

**Pull request merged**